Java

```diff
@@ -47,34 +58,27 @@ class student{
 		
 		
 		
-		
       	for(int t=0; t<w; t++) {
       	
           
         	System.out.print("Attendance for week # "+(t+1)+": ");
-        	time[t] = s.nextDouble(); 
-        
+        	double hours = s.nextDouble(); 
+        	students[i].addHours(t,hours);
+      	}
+	
         	
-      }
+  	
+        for(int t=0; t<n; t++){
 
-
-        
-        for(int t=0; t<w; t++){
-
-          total = total + time[t];
-
-        }          
-        double average = total / w;
```

Comment:
Please note: the calculation is incorrect: it should be `double average = total / w / 4;` (since it seems like a week of attendance = 4 hours, see line 73).